pandas/compat/pyarrow.py

```diff
@@ -14,9 +16,21 @@
     pa_version_under9p0 = _palv < Version("9.0.0")
     pa_version_under10p0 = _palv < Version("10.0.0")
     pa_version_under11p0 = _palv < Version("11.0.0")
+
+    PYARROW_INSTALLED = True
 except ImportError:
+    pa = None
+
     pa_version_under7p0 = True
     pa_version_under8p0 = True
     pa_version_under9p0 = True
     pa_version_under10p0 = True
     pa_version_under11p0 = True
+
+    PYARROW_INSTALLED = False
+
+
+def is_pyarrow_array(obj) -> bool:
```

Comment:
could make this a TypeGuard?

Reply:
I did an attempt to make it a TypeGuard (making it a TypeGuard when pyarrow is installed, and otherwise just annotate it with `bool`, not sure if putting a TypeGuard in a variable works this way)

Reply:
Thanks for taking a look